streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableMapValues.java

```diff
@@ -123,8 +130,11 @@ public void init(final ProcessorContext context) {
         }
 
         @Override
-        public V1 get(final K key) {
-            return computeValue(key, parentGetter.get(key));
+        public ValueAndTimestamp<V1> get(final K key) {
+            final ValueAndTimestamp<V> parentValueAndTimestamp = parentGetter.get(key);
+            final V parentValue = parentValueAndTimestamp == null ? null : parentValueAndTimestamp.value();
+            final V1 resultValue = computeValue(key, parentValue);
+            return resultValue == null ? null : ValueAndTimestamp.make(resultValue, parentValueAndTimestamp.timestamp());  // this is a potential NPE -- if `parentValueAndTimestamp == null` we could also directly return `null`, but this would be a semantic change
```

Comment:
Why can't we return `-1` if the timestamp is null like we did in other operators?

Reply:
The comment was for the old code. Not sure if it still applies.

however, the concern was to return `ValueAndTimestamp(null, -1)` instead of `null` -- I think, the former would be incorrect.